File: frappe_apps/www/custom_web_form.py
#:  Author  : Yuan-Yao Lou (Mike) <yylou@purdue.edu | yuanyao.lou@gmail.com>
#:  Title   : PhD in ECE at Purdue University
#:  Date    : 2021/06/08


from __future__ import unicode_literals

import frappe
from frappe.utils import now


@frappe.whitelist( allow_guest=True )
def get_response( first_name='', last_name='', email='', phone='', company='', job_title='', comments='' ):
    data = {
        'first name':   first_name,
        'last name':    last_name,
        'email':        email,
        'comments':     comments
    }

    #:  (for incomplete data, reply ERROR message)
    for key in data:
        if not data[key]: 
            frappe.response["message"] = 'Please provide your {0}, thanks.'.format( key )
            return frappe.response["message"]

    #:  (cyber attack protection)
    MAX_DATA_IN_AN_HOUR = 10000
    if frappe.db.sql("""select count(*) from `tabContact Data` where TIMEDIFF(%s, modified) < '01:00:00'""", now())[0][0] > MAX_DATA_IN_AN_HOUR:
        frappe.response["message"] = "Sorry, we believe we have received an unreasonably high number of requests of this kind. Please try later."
        return

    #:  TODO - (send an email to the consumer)
    # frappe.sendmail( recipients=forward_to_email, sender=sender, content='We have received your message and we will get back to you ASAP.', subject='[System Message] Reply Email from IoT Eye Inc.' )

    #:  (insert DocType data)
    frappe.get_doc( dict(
		doctype     = 'Web Form Data',
		first_name  = first_name,
        last_name   = last_name,
        email       = email,
        phone       = phone,
        company     = company,
        job_title   = job_title,
        comments    = comments,
	)).insert( ignore_permissions = True )

    return 'COMPLETE_DATA'
    

File: frappe_apps/www/custom_web_form.js
/**
 *	@Author	:	Yuan-Yao Lou (Mike) <yylou@purdue.edu | yuanyao.lou@gmail.com>
 *	@Title	:	PhD in ECE at Purdue University
 *	@Date	:	2021/06/08
 */


frappe.ready(function() {

	$('.btn-block').off( 'click' ).on( 'click', function() {
		/*  ---  Retrieve Input Data via 'name' HTML attritube  ---  */
		var first_name 	= $('[name="first_name"]').val();
		var last_name 	= $('[name="last_name"]').val();
		var email 		= $('[name="email"]').val();
        var phone		= $('[name="phone"]').val();
		var company		= $('[name="company"]').val();
        var job_title	= $('[name="job_title"]').val();
		var comments	= $('[name="comments"]').val();

		/*  ---  Debug Usage  ---  */
		console.log( first_name, ' | ', last_name, ' | ', email, ' | ', phone, ' | ', company, ' | ', job_title, ' | ', comments );

		frappe.call({
			method: 'frappe_apps.www.custom_web_form.get_response',
			args: {
				first_name:		first_name,
				last_name:		last_name,
				email:			email,
                phone:			phone,
				company:		company,
                job_title:		job_title,
				comments:		comments,
			},
			callback: function(r) {
				if ( r.message === 'COMPLETE_DATA' ) {
					/*  ---  Pop Up Message  ---  */
					frappe.msgprint( 'Thanks for your message.' );
					
					/*  ---  EMPTY all input fields  ---  */
					$(':input').val('');
				} else {
					frappe.msgprint( r.message );
				}
			}
		})
	})

});